python/cudf/dataframe/string.py

```diff
@@ -521,3 +521,29 @@ def _replace_defaults(self):
     def copy(self, deep=True):
         params = self._replace_defaults()
         return type(self)(**params)
+
+    def fillna(self, fill_value, inplace=False):
+        """
+        Fill null values with * fill_value *
+        """
+        result = self.copy()
+        if not isinstance(fill_value, str) and \
+            not(
+                isinstance(fill_value, series.Series) and
+                isinstance(fill_value._column, StringColumn)
+                ):
+            raise TypeError("fill_value must be a string or a string series")
+
+        # replace fill_value with nvstrings
+        # if it is a column
+
+        if isinstance(fill_value, series.Series):
+            if len(fill_value) < len(result):
```

Comment:
The behavior of numerical series seems to be incorrect here so I am not sure if we support replace with smaller length.  I am unsure on the expected behavior here. 
#### CUDF
```python
from cudf.dataframe import series
import pandas as pd
import cudf
df = cudf.from_pandas(pd.DataFrame({'id':[None,1,1,None]}))
repl = series.Series([-1,-1])
print(df['id'].fillna(repl))
```
Output
```python
0    nan
1    1.0
2    1.0
3    nan
Name: id, dtype: float64
```
#### Pandas 
```python
import pandas as pd
df = pd.DataFrame({'id':[None,1,1,None]})
repl = pd.Series([-1,-1])
print(df['id'].fillna(repl,inplace=False))

```
Output
```python
0   -1.0
1    1.0
2    1.0
3    NaN
```